File: javascript/문자열 여러 번 뒤집기.js
const solution = (my_string, queries) => {
    queries.forEach(a => {
        const copyArr = [...my_string];
        const [from, to] = a;
        const slicedText = copyArr.slice(from, to + 1);
        const reversedSlice = slicedText.reverse().join("");
        copyArr.splice(from, to - from+1, reversedSlice);
        my_string = copyArr.join("");
    })
    return my_string;
}

File: java/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        char[] chars = my_string.toCharArray();
        for(int[] query : queries) {
            int left = query[0];
            int right = query[1];
            while (left < right) {
                char c1 = chars[left];
                char c2 = chars[right];
                chars[left] = c2;
                chars[right] = c1;
                left++;
                right--;
            }
        }
        return new String(chars);
    }
}